reader_test.go

```diff
@@ -0,0 +1,40 @@
+package validator
+
+import (
+	"bytes"
+	"compress/flate"
+	"io"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// flateIt takes and input string, compresses it using flate, and returns a flate.Reader() of the compressed content
+func flateIt(t *testing.T, input string) io.Reader {
+	t.Helper()
+
+	var zipped bytes.Buffer
+	w, err := flate.NewWriter(&zipped, flate.DefaultCompression)
+	require.NoError(t, err)
+
+	w.Write([]byte(input))
+	w.Close()
```

Comment:
Let's also check for errors here for completeness.

Reply:
done (I force pushed)